### DISCUSSION TITLE
AssertSpecial Flag to Disable Roundstart Teleportation

### DISCUSSION THREAD

**Discussion by Kokorz** [ikemen-engine/Ikemen-GO#1161] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

In the current engine, there is no method of changing the behavior of round start without using hacky workarounds. By default, the engine automatically teleports players to their start coordinates when the round state reaches 2. This can cause issues for developers who want to customize the round start behavior for their games or game modes, as they are forced to use sub-optimal solutions.

### Describe the solution you'd like

I propose the addition of a new AssertSpecial flag that allows disabling the automatic round start teleportation for the character. This flag would enable developers to prevent players from being teleported to their start coordinates at the beginning of the round.

This flag could be named as "NoHardcodedRoundStartBehaviour"

### Additional context

This flag would allow people to easily create roundstart states that are similar to Melty Blood, where you can freely move around before the match begins.

### Search terms

roundstart teleportation teleport round start fight disable custom parameter customization assert flag assertspecial special

**Comment by potsmugen** [comment]
In my opinion that could just be flat out the normal behavior of the engine. If people want to move their characters in the intros, let them. They can already do it with workarounds anyway.

Otherwise maybe that should be a character or stage feature, not lifebar.

**Reply by Kokorz** [reply]
I'll be updating my request to be an assertspecial flag, I admit that would work better than a lifebar parameter

**Reply by potsmugen** [reply, marked as answer]
There's a new "NoIntroReset" flag in the nightly build, if you want to try it out.
